File: 169-majority-element/majority-element.cpp
// class Solution {
// public:
//     int majorityElement(vector<int>& nums) {
//         int winner=-1;
//         int count=0;
//         for(auto x:nums){
//             if(count==0){
//                 count=1;
//                 winner=x;
//             }else if(x==winner){
//                 count++;
//             }else{
//                 count--;
//             }
//         }
//         return winner;
//     }
// };



class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int,int>mp;
        for(int i=0;i<nums.size();i++){
            mp[nums[i]]++;
        }
        int ans=INT_MIN;
        int temp=INT_MIN;
        for(auto it:mp){
            if(it.second>temp){
                temp=it.second;
                ans=it.first;
            }
        }
        return ans;
    }
};

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums.sort()
        ans=1
        count=1
        val=0
        if len(nums)==1:
            return nums[0]
        for i in range(1,len(nums)):
            if nums[i-1]==nums[i]:
                count+=1
                if count>ans:
                    val=nums[i]
                ans=max(ans,count)
            else:
                count=1
        return val